### ISSUE TITLE
Http Client - GetAlias request throws class cast exception on missing index

### ISSUE THREAD

**Issue by muuki88** [sksamuel/elastic4s#824]
Hi,

When I run a `GetAlias` request and there are no aliases the client throws an exception instead of handling the empty response.

### expected behaviour

Either a failed Future, but I prefer a valid response with an empty Map.

### steps to reproduce

```scala
import com.sksamuel.elastic4s.http.HttpClient
import com.sksamuel.elastic4s.http.ElasticDsl
import com.sksamuel.elastic4s.http.ElasticDsl._
import scala.concurrent.duration.Duration

val httpClient: HttpClient = ???
val alias: Seq[String] = Seq("non-existing-alias")

// This call throws a parsing exception
val getAliasResponse = Await.result(httpClient.execute {
    getAlias(alias)
}, atMost = Duration(5, TimeUnit.SECONDS))

getAliasResponse.map {
    case (indexName, aliasesMap) =>
       val aliases = aliasesMap.getOrElse("aliases", Map.empty).keys
}.toSeq
```

If you call the rest endpoint directly with

```
_alias/non-existing
```

The response is

```json
{
  "error": "alias [non-existing-alias] missing",
  "status": 404
}
```

which the `GetAliasExecutable` is unable to parse, because the format is

```cala
type GetAliasResponse = Map[String, Map[String, Map[String, AnyRef]]]
```

**Comment by solicode** [comment]
I haven't looked too much into it, but this likely has to do with [this section](https://github.com/sksamuel/elastic4s/blob/4b80b9fa941d55d9e1bf067bcee2f6b25dfa8a36/elastic4s-http/src/main/scala/com/sksamuel/elastic4s/http/HttpClient.scala#L86-L94) I added fairly recently.

I made that change because there were many instances where exceptions were being thrown, when really the content itself was a "valid" response (for example, GetResponse where the `found: Boolean` flag on it is false). You'd never get that GetResponse because of the exception being thrown.

But as you can see, the implementation isn't complete. It fixed a few endpoints, but broke this one.

The way Elastic implemented the rest client makes handling this a bit difficult because I don't see a simple way of differentiating between these errors and "true" exceptions. I'm still not sure what the best way of handling this is. I wanted to avoid trying to parse twice, but I'm not sure what other options there are. I was also trying to avoid changing JsonFormat.

I can look into it more, but if anybody has some other ideas, I would appreciate it.

**Comment by sksamuel** [comment]
I think we might just have to handle it per endpoint in the HttpExecutable implementations. Sometimes a 404 means you've made an error, sometimes it's valid and means the document doesn't exist. And so on for other error codes. It's very inconsistent and bad API design.

**Comment by solicode** [comment]
That's a good idea. I thought it might lead to a lot of extra code, but it seems like there aren't too many different cases to consider. A lot of the code can be shared.

Since I left this half-broken and already know where all the occurrences of this are in the test, I went ahead an attempted a solution. I created the PR. Let me know what you think.

There might be other cases that there simply are no tests for currently, but I fixed everything that I could.
